### PR TITLE
fix: default import package.json for spec compatibility

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Fixes: https://github.com/aws/aws-sdk-js-v3/issues/1504
Replaces: https://github.com/aws/aws-sdk-js-v3/pull/1503

*Description of changes:*
default import package.json for spec compatibility

Special thanks to Co-author @rvaronos

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: clients/client-accessanalyzer/runtimeConfig.browser.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { name, version } from "./package.json";
+import * as packageInfo from "./package.json";
```
Comment (sokra) [inline clients/client-accessanalyzer/runtimeConfig.browser.ts:1]:
```suggestion
import packageInfo from "./package.json";
```

**Review by sokra — CHANGES_REQUESTED** [review]

File: clients/client-accessanalyzer/runtimeConfig.browser.ts (hunk 1 above)
Comment (trivikr) [reply clients/client-accessanalyzer/runtimeConfig.browser.ts:1]: @sokra Is there a reason for preferring default import over using whole module import (`* as`)?

Tested that version number is accessible in both default import as well as whole module import.
And the transpiled JavaScript is slightly smaller in size in whole module import.

<details>
<summary>package.json</summary>

```json
{
  "name": "import-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "typescript": "4.0.2"
  }
}
```

</details>

<details>
<summary>Default Import</summary>

```console
$ echo "import packageInfo from './package.json';                              
console.log(packageInfo.version);" > defaultImport.ts

$ ./node_modules/.bin/tsc --resolveJsonModule --esModuleInterop defaultImport.ts

$ node defaultImport.js 
1.0.0

$ cat defaultImport.js 
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var package_json_1 = __importDefault(require("./package.json"));
console.log(package_json_1["default"].version);
```

</details>

<details>
<summary>Whole module import</summary>

```console
$ echo "import * as packageInfo from './package.json';                          
console.log(packageInfo.version);" > wholeModuleImport.ts

$ ./node_modules/.bin/tsc --resolveJsonModule wholeModuleImport.ts

$ node wholeModuleImport.js 
1.0.0

$ cat wholeModuleImport.js 
"use strict";
exports.__esModule = true;
var packageInfo = require("./package.json");
console.log(packageInfo.version);
```

</details>

Comment (trivikr) [reply clients/client-accessanalyzer/runtimeConfig.browser.ts:1]: Verified that webpack@beta doesn't show warning on transpiled JavaScript

<details>
<summary>Output</summary>

```console
$ ./node_modules/.bin/webpack --version            
webpack-cli 1.0.0
webpack 5.0.0-beta.29

$ ./node_modules/.bin/webpack --target=node --entry defaultImport.js    

✅ Compilation Results

🌏 Version: 5.0.0-beta.29
⚒️  Built: Wed Sep 09 2020 09:40:31 GMT-0700 (Pacific Daylight Time)
⏱  Compile Time: 404ms
📂 Output Directory: /Users/trivikr/workspace/import-test/dist
#️⃣ Hash: 8ce8943292a079a2e100

┌────────────┬──────────────────────────────────────┐
│ Entrypoint │ Bundle                               │
├────────────┼──────────────────────────────────────┤
│ main       │  Bundle Name       │ main.js         │
│            │  Compared For Emit │ false           │
│            │  Bundle size       │ 0.64 kb         │
│            │                                      │
│            │ Modules:                             │
│            │ ./defaultImport.js [size: 293 bytes] │
│            │ ./package.json [size: 288 bytes]     │
└────────────┴──────────────────────────────────────┘

[webpack-cli] configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/

$ ./node_modules/.bin/webpack --target=node --entry wholeModuleImport.js 

✅ Compilation Results

🌏 Version: 5.0.0-beta.29
⚒️  Built: Wed Sep 09 2020 09:41:06 GMT-0700 (Pacific Daylight Time)
⏱  Compile Time: 289ms
📂 Output Directory: /Users/trivikr/workspace/import-test/dist
#️⃣ Hash: c0f30dea06da7cdcb856

┌────────────┬──────────────────────────────────────────┐
│ Entrypoint │ Bundle                                   │
├────────────┼──────────────────────────────────────────┤
│ main       │  Bundle Name       │ main.js             │
│            │  Compared For Emit │ false               │
│            │  Bundle size       │ 0.51 kb             │
│            │                                          │
│            │ Modules:                                 │
│            │ ./package.json [size: 288 bytes]         │
│            │ ./wholeModuleImport.js [size: 120 bytes] │
└────────────┴──────────────────────────────────────────┘

[webpack-cli] configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/
```

</details>

Comment (sokra) [reply clients/client-accessanalyzer/runtimeConfig.browser.ts:1]: If you transpile it to CommonJs it doesn't matter for webpack anyway and you can keep the original code.

The webpack warning refers to a spec proposal where the common agreement is that a JSON module has only a default export. This makes `import { name } from "package.json"` an Error in native ESM.

Comment (trivikr) [reply clients/client-accessanalyzer/runtimeConfig.browser.ts:1]: We'll switch to default imports for other reasons as explained in https://github.com/aws/aws-sdk-js-v3/issues/1511

**Comment by trivikr** [comment]
This PR would be set to ready for review once https://github.com/awslabs/smithy-typescript/pull/209 is ready

File: clients/client-accessanalyzer/runtimeConfig.browser.ts (hunk 1 above)
Comment (trivikr) [reply clients/client-accessanalyzer/runtimeConfig.browser.ts]: packageInfo is now default imported in [`d2ee7ca` (#1505)](https://github.com/aws/aws-sdk-js-v3/pull/1505/commits/d2ee7ca3d7192c1ef4547e0e6d0bdaccb481a273)

**Review by AllanFly120 — APPROVED** [review]

**Pull request merged**